gem5, m5ops: init at 24.1.0.1

mtoohey31 opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Closes #348629.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mtoohey31 committed:
gem5, m5ops: init at 24.1.0.3

ethancedwards8 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 375739`

Arya-Elfren commented:
Nice! Thanks for packaging this for me (got locked out of my GH account for a bit, sorry). This builds for me (and so I can use it). Would love for it to get merged though.

ethancedwards8 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -18352,6 +18352,8 @@ with pkgs;
     erlang = erlang_25;
   };
 
+  m5ops = callPackage ../by-name/ge/gem5/m5ops.nix { };
```
I don't think this is necessary anymore.

mtoohey31 replied:
Just tried removing it and I get a missing attribute error. I think it's because I'm using a non-standard name within the gem5 parent directory. I put the package in there cause it's very closely related and should pretty much always be used with gem5 even though the builds are separate.

Should I put it in its own directory or is it fine to leave this?

ethancedwards8 replied:
Yeah, you're right, it needs to be package.nix. To be honest, I'm not sure. It probably needs its own directory, but someone more knowledgeable than me should answer that.

mtoohey31 replied:
:+1: ok. If you search for `by-name.*\{ \}` in `all-packages.nix` there are a couple other examples of this, so I'll leave it for now, but if someone has more information about the correct layout lmk.

mtoohey31 commented:
I've resolved the merge conflict. If you want to make this more visible to maintainers, you can add a :+1:.

pbsds commented on pkgs/by-name/ge/gem5/m5ops.nix:
please move this to pkgs/by-name/m5/m5ops/package.nix and remove the entry from all-packages.nix.

mtoohey31 replied:
Done in 120221b7da9a6f56c0ea3187faa82a3bc41d1787.

pbsds commented on pkgs/by-name/ge/gem5/m5ops.nix:
```diff
@@ -0,0 +1,53 @@
+{
+  fetchFromGitHub,
+  glibc,
+  isa ? "x86",
```
I assume this is the simulation target? Are other ISAs available? please list them in a comment for both packages if yes

mtoohey31 commented:
Thanks for taking a look @pbsds! I believe I've addressed both of your pieces of feedback. Just a note though: in that last commit I also removed myself as maintainer for both packages, as I have stopped using them in the 5 months since I originally created the pull request. If anyone else would like to volunteer to maintain them, please let me know.
